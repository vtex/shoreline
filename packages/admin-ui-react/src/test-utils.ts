import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
