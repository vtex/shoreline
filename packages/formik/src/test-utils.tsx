import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
