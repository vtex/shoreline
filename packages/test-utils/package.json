{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
