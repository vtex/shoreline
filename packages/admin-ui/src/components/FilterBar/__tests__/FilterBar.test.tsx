import React from 'react'
import { render, axe } from '../../../test-utils'

import { FilterBar } from '../index'
import { useFilterBarState } from '../useFilterBarState'
import type {
  UseFilterBarStateParams,
  UseFilterBarStateReturn,
} from '../typings'

interface FilterStateProps<T> extends UseFilterBarStateParams<T> {
  children: (state: UseFilterBarStateReturn<T>) => JSX.Element
}

function StatefulFilterBar<T, V extends { value: T }>({
  children,
  ...hookProps
}: FilterStateProps<V>) {
  const state = useFilterBarState({ ...hookProps })

  return children(state)
}

describe('FilterBar tests', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    })
  })

  const internalLabels = {
    conjunctionLabel: 'Conjunction',
    filterLabel: 'Filter',
    conditionLabel: 'Condition',
    statementMenuLabel: 'Statement Menu',
    applyFilterLabel: 'Apply',
    addFilterLabel: 'Add Filter',
    clearFilterLabel: 'Clear Filters',
    deleteStatementLabel: 'Delete',
    duplicateStatementLabel: 'Duplicate',
    whereStatementLabel: 'Where',
  }

  it('should not have a11y violations', async () => {
    const { container } = render(
      <>
        <StatefulFilterBar
          filters={[
            {
              label: 'Filter',
              id: 'filter',
              conditions: [{ label: 'is', id: '1' }],
              resolver: {
                type: 'simple',
                defaultValue: { value: 'item 1' },
                items: [
                  { value: 'item 1' },
                  { value: 'item 2' },
                  { value: 'item 3' },
                ],
              },
            },
          ]}
          onApply={() => {}}
          statements={[
            {
              condition: { label: 'is', id: '1' },
              filter: {
                label: 'Filter',
                id: 'filter',
                conditions: [{ label: 'is', id: '1' }],
                resolver: {
                  type: 'simple',
                  defaultValue: { value: 'item 1' },
                  items: [
                    { value: 'item 1' },
                    { value: 'item 2' },
                    { value: 'item 3' },
                  ],
                },
              },
              target: { value: 'item 1' },
            },
          ]}
          conjunction={{ label: 'And', value: 'and' }}
        >
          {(state) => (
            <FilterBar
              state={state}
              label="Filter Bar"
              csx={{ bg: 'coral' }}
              data-testid="filter-bar"
              conjunctions={[
                { label: 'And', value: 'and' },
                { label: 'Or', value: 'or' },
              ]}
              internalLabels={internalLabels}
            />
          )}
        </StatefulFilterBar>
      </>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
