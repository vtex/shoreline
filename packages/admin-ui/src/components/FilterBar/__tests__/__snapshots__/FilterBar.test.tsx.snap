// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-51 {
  border: 1px solid hsla(0, 0%, 87%, 1);
  background: coral;
  border-radius: 4px;
}

.emotion-50 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-top: 1px solid hsla(0, 0%, 87%, 1);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  background: container;
}

.emotion-42 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 32px;
  width: auto;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
  color: hsla(222, 63%, 43%, 1);
  background: transparent;
}

.emotion-42:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-42:hover {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(224, 65%, 97%, 1);
}

.emotion-42:active {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(222, 62%, 92%, 1);
}

.emotion-42:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-42:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-42 svg {
  margin: 0.25rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-23 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.emotion-49 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
}

.emotion-49 >*:not(:first-child) {
  margin-left: 0.25rem;
}

.emotion-45 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 32px;
  width: auto;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  color: #000000;
  background: transparent;
}

.emotion-45:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-45:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-45:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-45:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-45:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-45 svg {
  margin: 0.25rem;
}

.emotion-48 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 32px;
  width: auto;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
}

.emotion-48:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-48:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-48:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-48:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-48:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-48 svg {
  margin: 0.25rem;
}

.emotion-38 {
  padding: 1rem;
  padding-left: 1rem;
}

.emotion-37 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-37 >*:not(:last-child) {
  margin-bottom: 0.5rem;
}

.emotion-36 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
}

.emotion-21 >*:not(:first-child) {
  margin-left: 0.5rem;
}

.emotion-0 {
  padding-left: 0.75rem;
  min-width: 100px;
  max-width: 100px;
}

.emotion-7 {
  position: relative;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 0.75rem;
  padding-right: 0.5rem;
  color: #000000;
  background: transparent;
  min-width: 150px;
  max-width: 150px;
}

.emotion-5:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-5:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-5:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-5 svg {
  margin: 0.25rem;
}

.emotion-5 div {
  width: 100%;
}

.emotion-5 div div {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-2 {
  text-align: start;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
  visibility: hidden;
  background: #FFFFFF;
  border: 1px solid hsla(0, 0%, 87%, 1);
  border-radius: 4px;
  box-shadow: 0rem 0rem 1rem 0rem rgba(0,0,0,0.10);
  cursor: pointer;
  outline: none;
  margin-top: 0.25rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  min-width: inherit;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: absolute;
  z-index: 999;
}

.emotion-6 >*:not(:last-child) {
  margin-bottom: 0.5rem;
}



.emotion-27 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 1rem;
  padding-right: 1rem;
  color: #000000;
  background: transparent;
}

.emotion-27:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-27:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-27:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-27:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-27:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-27 svg {
  margin: 0.25rem;
}

.emotion-27:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-27:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-27:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-27:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-27:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-27 svg {
  margin: 0.25rem;
}

.emotion-34 {
  outline: none;
  z-index: 999;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FFFFFF;
  padding: 0.75rem;
  min-width: 18px;
  border: 1px solid hsla(0, 0%, 87%, 1);
  border-radius: 3px;
  box-shadow: 0rem 0rem 1rem 0rem rgba(0,0,0,0.10);
}

.emotion-30 {
  margin-top: 0.125rem;
  margin-bottom: 0.125rem;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0071em;
  border: none;
  text-transform: initial;
  width: 100%;
  height: 32px;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  background: transparent;
  color: hsla(222, 63%, 43%, 1);
}

.emotion-30 div {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-30 svg {
  margin: 0.25rem;
  width: 20px;
  height: 20px;
  min-width: 20px;
  min-height: 20px;
  margin-left: 0rem;
  margin-right: 0.5rem;
}

.emotion-30:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-30:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-30:hover {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(224, 65%, 97%, 1);
}

.emotion-30:active {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(222, 62%, 92%, 1);
}

.emotion-30:disabled {
  color: hsla(0, 0%, 43%, 1);
  background: hsla(0, 0%, 87%, 1);
}

.emotion-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-28 {
  width: 1rem;
  height: 1rem;
  min-height: 1rem;
  min-width: 1rem;
}

<div
    class="emotion-51"
    data-testid="filter-bar"
  >
    <div
      class="emotion-38"
    >
      <div
        class="emotion-37"
      >
        <div
          class="emotion-36"
        >
          <div
            class="emotion-21"
          >
            <div
              class="emotion-0"
            >
              Where
            </div>
            <div
              class="emotion-7"
            >
              <span
                style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
              >
                <label
                  for="downshift-0-toggle-button"
                  id="downshift-0-label"
                >
                  Filter
                </label>
              </span>
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-0-label downshift-0-toggle-button"
                class="emotion-5"
                id="downshift-0-toggle-button"
                type="button"
              >
                <div
                  class="emotion-4"
                >
                  <div
                    class="emotion-3"
                  >
                    <svg
                      class="emotion-1"
                      fill="currentColor"
                      height="1.25rem"
                      viewBox="0 0 256 256"
                      width="1.25rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="16rem"
                        width="16rem"
                      />
                      <polyline
                        fill="none"
                        points="208 96 128 176 48 96"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="18"
                      />
                    </svg>
                     
                    <div
                      class="emotion-2"
                    >
                      Filter
                    </div>
                  </div>
                </div>
              </button>
              <div
                aria-labelledby="downshift-0-label"
                class="emotion-6"
                id="downshift-0-menu"
                role="listbox"
                tabindex="-1"
              />
            </div>
            <div
              class="emotion-7"
            >
              <span
                style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
              >
                <label
                  for="downshift-1-toggle-button"
                  id="downshift-1-label"
                >
                  Condition
                </label>
              </span>
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-1-label downshift-1-toggle-button"
                class="emotion-5"
                id="downshift-1-toggle-button"
                type="button"
              >
                <div
                  class="emotion-4"
                >
                  <div
                    class="emotion-3"
                  >
                    <svg
                      class="emotion-1"
                      fill="currentColor"
                      height="1.25rem"
                      viewBox="0 0 256 256"
                      width="1.25rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="16rem"
                        width="16rem"
                      />
                      <polyline
                        fill="none"
                        points="208 96 128 176 48 96"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="18"
                      />
                    </svg>
                     
                    <div
                      class="emotion-2"
                    >
                      is
                    </div>
                  </div>
                </div>
              </button>
              <div
                aria-labelledby="downshift-1-label"
                class="emotion-6"
                id="downshift-1-menu"
                role="listbox"
                tabindex="-1"
              />
            </div>
            <div
              class="emotion-7"
            >
              <span
                style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
              >
                <label
                  for="downshift-2-toggle-button"
                  id="downshift-2-label"
                >
                  Value
                </label>
              </span>
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="downshift-2-label downshift-2-toggle-button"
                class="emotion-5"
                id="downshift-2-toggle-button"
                type="button"
              >
                <div
                  class="emotion-4"
                >
                  <div
                    class="emotion-3"
                  >
                    <svg
                      class="emotion-1"
                      fill="currentColor"
                      height="1.25rem"
                      viewBox="0 0 256 256"
                      width="1.25rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="16rem"
                        width="16rem"
                      />
                      <polyline
                        fill="none"
                        points="208 96 128 176 48 96"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="18"
                      />
                    </svg>
                     item 1
                  </div>
                </div>
              </button>
              <div
                aria-labelledby="downshift-2-label"
                class="emotion-6"
                id="downshift-2-menu"
                role="listbox"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            class="emotion-35"
          >
            <button
              aria-controls="statement-menu"
              aria-expanded="false"
              aria-haspopup="menu"
              aria-label="Statement Menu 0"
              class="emotion-27"
              type="button"
            >
              <div
                class="emotion-4"
              >
                <div
                  class="emotion-23"
                >
                  <svg
                    class="emotion-1"
                    fill="currentColor"
                    height="1.25rem"
                    viewBox="0 0 256 256"
                    width="1.25rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="16rem"
                      width="16rem"
                    />
                    <circle
                      cx="128"
                      cy="64"
                      r="16"
                    />
                    <circle
                      cx="128"
                      cy="128"
                      r="16"
                    />
                    <circle
                      cx="128"
                      cy="192"
                      r="16"
                    />
                  </svg>
                   
                  <div
                    class="emotion-4"
                  >
                    <div
                      class="emotion-23"
                    >
                       
                    </div>
                  </div>
                </div>
              </div>
            </button>
            <div
              aria-label="Statement Menu 0"
              aria-orientation="vertical"
              class="emotion-34"
              data-dialog="true"
              hidden=""
              id="statement-menu"
              role="menu"
              style="display: none; position: fixed; left: 100%; top: 100%;"
              tabindex="-1"
            >
              <button
                class="emotion-30"
                id="statement-menu-1"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="emotion-29"
                >
                  <svg
                    class="emotion-28"
                    fill="currentColor"
                    height="1rem"
                    viewBox="0 0 256 256"
                    width="1rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="16rem"
                      width="16rem"
                    />
                    <polyline
                      fill="none"
                      points="168 167.993 216 167.993 216 39.993 88 39.993 88 87.993"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="18"
                    />
                    <rect
                      fill="none"
                      height="128"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="18"
                      width="128"
                      x="39.99902"
                      y="87.99414"
                    />
                  </svg>
                   Duplicate
                </div>
              </button>
              <button
                class="emotion-30"
                id="statement-menu-2"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                <div
                  class="emotion-29"
                >
                  <svg
                    class="emotion-28"
                    fill="currentColor"
                    height="1rem"
                    viewBox="0 0 256 256"
                    width="1rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="16rem"
                      width="16rem"
                    />
                    <line
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="18"
                      x1="215.99609"
                      x2="39.99609"
                      y1="56"
                      y2="56.00005"
                    />
                    <line
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="18"
                      x1="104"
                      x2="104"
                      y1="104"
                      y2="168"
                    />
                    <line
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="18"
                      x1="152"
                      x2="152"
                      y1="104"
                      y2="168"
                    />
                    <path
                      d="M200,56V208a8,8,0,0,1-8,8H64a8,8,0,0,1-8-8V56"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="18"
                    />
                    <path
                      d="M168,56V40a16,16,0,0,0-16-16H104A16,16,0,0,0,88,40V56"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="18"
                    />
                  </svg>
                   Delete
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-50"
    >
      <button
        class="emotion-42"
        type="button"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-23"
          >
            <svg
              class="emotion-1"
              fill="currentColor"
              height="1.25rem"
              viewBox="0 0 256 256"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16rem"
                width="16rem"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="40"
                x2="216"
                y1="128"
                y2="128"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
                x1="128"
                x2="128"
                y1="40"
                y2="216"
              />
            </svg>
             Add Filter
          </div>
        </div>
      </button>
      <div
        class="emotion-49"
      >
        <button
          class="emotion-45"
          type="button"
        >
          <div
            class="emotion-4"
          >
            <div
              class="emotion-23"
            >
               Clear Filters
            </div>
          </div>
        </button>
        <button
          aria-disabled="true"
          class="emotion-48"
          disabled=""
          style="pointer-events: none;"
          type="button"
        >
          <div
            class="emotion-4"
          >
            <div
              class="emotion-23"
            >
               Apply
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
